pkg/monitor/monitor/monitor_manager.go

```diff
@@ -224,15 +224,18 @@ func (m *MonitorManager) syncTidbMonitorStatefulset(tc *v1alpha1.TidbCluster, dc
 		return err
 	}
 
-	result, err := m.smoothMigrationToStatefulSet(monitor)
-	if err != nil {
-		klog.Errorf("Fail to migrate from deployment to statefulset for tm [%s/%s], err: %v", ns, name, err)
-		return err
-	}
-	if !result {
-		klog.Infof("Wait for the smooth migration to be done successfully for tm [%s/%s]", ns, name)
-		return nil
+	if monitor.Status.DeploymentStorageStatus != nil {
```

Comment:
Did you test the code change here? 
When will `DeploymentStorageStatus != nil` and trigger the migration now?
With this change, can the deployment in 1.1 be migrated to sts after this change?
Could you please update the test result in the PR?

Reply:
I think only when smooth migrate will sync this DeploymentStorageStatus, there will not trigger migration if adding this justify.

Reply:
I did not expect that `DeploymentStorageStatus` is a new field 😂 This `if` should be removed.